src/bin/cargo-msrv.rs

```diff
@@ -18,20 +18,25 @@ use cargo_msrv::{run_app, Context, OutputFormat, TracingOptions, TracingTargetOp
 
 fn main() {
     std::process::exit(
-        match _main(std::env::args_os) {
+        match _main_1(std::env::args_os) {
             Ok((_guard, exit_code)) => exit_code,
             Err(err) => {
-                tracing::error!("{}", err);
+                // Don't use `tracing::error!` here because maybe an issue with `tracing` setup is what
+                // caused this error in the first place
+                // Additionally `tracing::error!` would not print to console by default, so user would not
+                // know what went wrong
+                println!("Setup error: {}", err);
                 ExitCode::Failure
             }
         }
         .into(),
     );
 }
 
-fn _main<I: IntoIterator<Item = OsString>, F: FnOnce() -> I + Clone>(
+/// Step 1: Parse options and set up tracing
+fn _main_1<I: IntoIterator<Item = OsString>, F: FnOnce() -> I + Clone>(
```

Comment:
I have chosen the `_main_1`, `_main_2` and `_main_3` names just for the first proof-of-concept. Please let me know if you have ideas for better names, or if I should merge some of the functions.